frontend/src/core/api/user.js

```diff
@@ -102,4 +102,14 @@ export default class UserAPI extends APIBase {
             headers,
         );
     }
+
+    /**
+     * Dismiss Add-On Promotion.
+     */
+    dismissAddonPromotion(): Promise<any> {
+        const headers = new Headers();
+        headers.append('X-Requested-With', 'XMLHttpRequest');
+
+        return this.fetch('/dismiss-addon-promotion/', 'GET', null, headers);
```

Comment:
For anything REST-like, this request should definitely be `POST` or even better `PATCH`, and should go to the same endpoint the information is retrieved from. But if we're not building an API here, I think that's fine.

Reply:
One idea, should Pontoon Add-on itself call this endpoint upon installation or regularly with fetching new data to dismiss the promotion even before the user visits any page?

If that would be nice to do, I would advise not to call it `/dismiss-addon-promotion/`, but rather `/pontoon-addon-installed/` and POST the JSON object equivalent to what's now injected into the `window` there.

Reply:
Our API story is a joke. :) I should make that request POST at least, I'd leave the rest for a followup.

Regarding the Add-on calling into that API I'm not sure. We have `postMessage` in place that effectively serves the same purpose, and use the `dismissed` flag to track which user has explicitly dismissed the promotion. Perhaps we should keep these actions separate.